<a href="https://colab.research.google.com/github/irssaa29/Machine-learning_01/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/irssaa29/Machine-learning_01/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*
I am choosing Lane 2: Refresh / Content Opportunity Scoring. In my Week 1 notebooks, I found that a simple hand-written rule (stale content that's still visible) scored Precision@50 = 0.680 — beating a shallow decision tree (0.600) and only getting matched by a much deeper, harder-to-read tree. That tells me there's real, learnable signal in this dataset worth building into a full scoring system, and it's a natural extension of the comparison work I already did.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*
Decision: Which pages should a content reviewer prioritize for refresh, given limited review time?
Unit of analysis: One row = one content page.
Action: A reviewer opens the top-ranked pages from my output and updates, expands, or protects them.
Cost of a wrong call: If I rank a healthy page as "needs review," the reviewer wastes limited time on a page that didn't need it. If I miss a genuinely declining page, it keeps losing visibility unnoticed — a real business cost, since traffic decline compounds the longer it goes unaddressed.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*
Here are real numbers from the starter dataset that support this lane.

In [12]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")
df["is_declining_label"] = df["trend_direction"].str.lower().eq("down").astype(int)

print(df.shape)
declining_rate = df["is_declining_label"].mean()
median_impressions = df["impressions_90d"].median()
stale_and_visible = ((df["days_since_last_update"] >= 180) & (df["impressions_90d"] >= 500)).sum()

print(f"Share of pages currently declining: {declining_rate:.3f}")
print(f"Median 90-day impressions: {median_impressions:.0f}")
print(f"Pages that are stale AND still visible: {stale_and_visible} out of {len(df)}")

(30000, 45)
Share of pages currently declining: 0.542
Median 90-day impressions: 731
Pages that are stale AND still visible: 17 out of 30000


Of the 30,000 pages in the starter dataset, 54.2% are currently labeled as declining, and the median page still receives 731 impressions over 90 days — so this is a dataset with real visibility at stake, not abandoned content. However, only 17 pages meet the strict "stale AND visible" rule from Week 1, showing that a single narrow baseline won't surface enough candidates on its own. This confirms Lane 2 is worth pursuing, but also tells me my baseline will need more than one rule (e.g. combining "declining with demand" and "page-one decay risk" from the lane guide) to build a useful, appropriately-sized review queue.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

What this work can say: Based on the starter dataset, I can identify pages that are observed to be declining, stale, or under-visible using transparent, explainable rules and simple models — and I can rank them so a reviewer knows where to look first. Any relationship I find between signals (like staleness, impressions, or position) and decline is associative and directional — it describes a pattern in the data, not a guaranteed cause. My output is decision-support: a ranked list with reasons attached, meant to guide a human reviewer's limited time, not replace their judgment.

What this work can never say: I cannot claim that refreshing a page causes it to recover — proving that would require an actual experiment (e.g. refreshing some pages and comparing them to similar pages left untouched), which this dataset does not provide. I also cannot claim to have discovered anything about how Google's search algorithm actually works — I'm only observing search performance signals, not the ranking system itself. A high score from my model means "worth reviewing first," never "guaranteed to improve."

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.